sentinelhub/api/batch/process.py

```diff
@@ -29,6 +31,23 @@
 BatchCollectionType = Union[str, dict, "BatchCollection"]
 
 
+def batch_user_action_wait_for_status_change(func: Callable) -> Callable:
+    @wraps(func)
+    def retrying_func(self: SentinelHubBatch, batch_request: BatchRequest) -> Json:
```

Comment:
@zigaLuksic not sure about the `self` type here. It works for batch V1, but batch V2 in principle has a different class, which I don't import here, but `mypy` seems to be happy